src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -156,14 +157,12 @@ public Response createDatasetRegistration(
             FormDataMultiPart multipart,
             @FormDataParam("dataset") String json) {
         try {
-            try {
-                if (!jsonSchemaUtil.isValidSchema_v1(json)) {
-                    throw new BadRequestException("Invalid schema");
-                }
-            } catch (Exception e) {
-                throw new BadRequestException("Invalid schema");
+            List<String> errors = jsonSchemaUtil.validateSchema_v1(json);
+            if (!errors.isEmpty()) {
+                throw new BadRequestException("Invalid schema:\n" + String.join("\n", errors));
             }
```

Comment:
Optional - we have an `Error` record class meant for this kind of thing and works well when converted to json: `org.broadinstitute.consent.http.models.Error`

Reply:
Good idea. I'll change that.